### PR TITLE
[API] Fix service path resolution

### PR DESCRIPTION
**Pull request by liranbg**
the previous regex captured wrong paths, e.g causing `projects/fraud-demo-admin/nuclio/events-ingest/deploy` to be captured as alerts service falsely.

added a test to cover all endpoints and ensure routers with specific tags are redirected to alerts service

https://iguazio.atlassian.net/browse/ML-8793

### CODE REVIEW

File: server/py/framework/utils/clients/discovery.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def _resolve_service_method_routes(
             "put": [],
             "delete": [],
             "patch": [],
+            "head": [],
```
Comment (moranbental) [inline server/py/framework/utils/clients/discovery.py:99]: Why do we need that here? I haven't seen a request type like that in our code

**Review by moranbental — CHANGES_REQUESTED** [review]
I think it’s worth adding this log:
`logger.debug("Skipping DB session creation for forwarded request", method=request.method, path=request.url.path)
` here:
https://github.com/mlrun/mlrun/blob/449735b0a857bc03e3db1f9fe37063af1fc367d6/server/py/framework/api/deps.py#L42
That way, we can identify the problem next time if it happens.

File: server/py/framework/utils/clients/discovery.py (hunk 1 above)
Comment (liranbg) [reply server/py/framework/utils/clients/discovery.py:99]: needed for docs endpoint

**Comment by liranbg** [comment]
> logger.debug("Skipping DB session creation for forwarded request", method=request.method, path=request.url.path)

this is too spammy and we dont have verbose log level

**Review by moranbental — APPROVED** [review]
LGTM

**Pull request merged**